File: spring/src/main/kotlin/com/windsnow1025/leximemo/spring/entity/Dictionary.kt
package com.windsnow1025.leximemo.spring.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*

@Entity
class Dictionary(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int? = null,
    @Column(unique = true)
    var name: String,

    @ManyToMany
    @JoinTable(
        name = "dictionary_word",
        joinColumns = [JoinColumn(name = "dictionary_id")],
        inverseJoinColumns = [JoinColumn(name = "word_id")]
    )
    @JsonBackReference
    var words: MutableSet<Word> = mutableSetOf()
)

File: next/src/model/Dictionary.ts
import {Word} from "@/model/Word";

export class Dictionary {
  id?: number;
  name: string = '';
  words: Word[] = [];

  constructor(id?: number, name: string = '', words: Word[] = []) {
    this.id = id;
    this.name = name;
    this.words = words;
  }
}
